src/lib.rs

```diff
@@ -304,6 +344,11 @@ pub fn store_path<T: Serialize>(path: impl AsRef<Path>, cfg: T) -> Result<(), Co
         .open(path)
         .map_err(ConfyError::OpenConfigurationFileError)?;
 
+    if let Some(p) = perms {
+        f.set_permissions(p)
+            .map_err(ConfyError::OpenConfigurationFileError)?;
```

Comment:
I guess this is not an `OpenConfigurationFileError` but a `SetPermissionsFileError`